generator/default.p9k

```diff
@@ -407,8 +421,8 @@ function __p9k_async_callback() {
 
   # split ${RAW_SEGMENT_DATA} into an array - see https://unix.stackexchange.com/a/28873
   local segment_meta=("${(@s:·|·:)RAW_SEGMENT_DATA}") # split on delimiter "·|·" (@s:<delim>:)
-  local cache_key="${segment_meta[2]}::${segment_meta[3]}"
-  __p9k_segment_cache["${cache_key}"]="${RAW_SEGMENT_DATA}"
+  local cache_key="${segment_meta[3]}::${segment_meta[1]}"
+  __p9k_segment_cache["${cache_key}"]="${RAW_SEGMENT_DATA#*·|·}"
```

Comment:
Hmm. Is cutting off the segment name in the cache entry intentional?

Reply:
Yes. Because "segment name" may be different with "STATEFUL_NAME". I added a "${segment}·|·" for the `__p9k_async_wrapper` only, in order the segment name could be retrieved in callback. In callback, we need to cut the additional segment name prefix in output.